clang/include/clang/CConv/ConstraintVariables.h

```diff
@@ -311,7 +305,9 @@ class PointerVariableConstraint : public ConstraintVariable {
 
   bool getIsGeneric() const { return IsGeneric; }
 
-  bool getIsOriginallyChecked() const override { return OriginallyChecked; }
+  bool getIsOriginallyChecked() const override {
+    return llvm::any_of(vars, [](Atom *A) { return isa<ConstAtom>(A); });
+  }
```

Comment:
This is a clever encoding. I am wondering: Now that our notion of "it was checked in the input program" is more evolved, should the callers of this method be reconsidered too? They may have an oversimple view of a variable being checked in the input program.

Reply:
The consequential use of this method is in `link`, so it would effect extern functions I think. 

Looking at this example, I think it will need to be changed.

```c
extern void test(_Ptr<int *> a);
void foo() {
  int ** b = 0;
  // Converts to: _Ptr<_Ptr<int> b = 0;
  test(b);
}
```

It makes `b` full checked even though `test` is extern with the inner pointer unchecked.